packages/babel-plugin-proposal-class-properties/src/index.js

```diff
@@ -1,436 +1,22 @@
-              ]),
-            );
-            computedNode.key = t.cloneNode(ident);
-          }
-        }
-
-        // Transform private props before publics.
-        const privateMaps = [];
-        const privateMapInits = [];
-        for (const prop of props) {
-          if (prop.isPrivate()) {
-            const inits = [];
-            privateMapInits.push(inits);
-
-            privateMaps.push(
-              buildClassPrivateProperty(t.thisExpression(), prop, inits, state),
-            );
-          }
-        }
-
-        let p = 0;
-        for (const prop of props) {
-          if (prop.node.static) {
-            staticNodes.push(buildClassProperty(t.cloneNode(ref), prop, state));
-          } else if (prop.isPrivate()) {
-            instanceBody.push(privateMaps[p]());
-            staticNodes.push(...privateMapInits[p]);
-            p++;
-          } else {
-            instanceBody.push(
-              buildClassProperty(t.thisExpression(), prop, state),
-            );
-          }
-        }
-
-        if (instanceBody.length) {
-          if (!constructor) {
-            const newConstructor = t.classMethod(
-              "constructor",
-              t.identifier("constructor"),
-              [],
-              t.blockStatement([]),
-            );
-            if (isDerived) {
-              newConstructor.params = [t.restElement(t.identifier("args"))];
-              newConstructor.body.body.push(
-                t.expressionStatement(
-                  t.callExpression(t.super(), [
-                    t.spreadElement(t.identifier("args")),
-                  ]),
-                ),
-              );
-            }
-            [constructor] = body.unshiftContainer("body", newConstructor);
-          }
-
-          const state = { scope: constructor.scope };
-          for (const prop of props) {
-            if (prop.node.static) continue;
-            prop.traverse(referenceVisitor, state);
-          }
-
-          //
-
-          if (isDerived) {
-            const bareSupers = [];
-            constructor.traverse(findBareSupers, bareSupers);
-            for (const bareSuper of bareSupers) {
-              bareSuper.insertAfter(instanceBody);
-            }
-          } else {
-            constructor.get("body").unshiftContainer("body", instanceBody);
-          }
-        }
-
-        for (const prop of props) {
-          prop.remove();
-        }
-
-        if (computedNodes.length === 0 && staticNodes.length === 0) return;
-
-        if (path.isClassExpression()) {
-          path.scope.push({ id: ref });
-          path.replaceWith(
-            t.assignmentExpression("=", t.cloneNode(ref), path.node),
-          );
-        } else if (!path.node.id) {
-          // Anonymous class declaration
-          path.node.id = ref;
-        }
-
-        path.insertBefore(computedNodes);
-        path.insertAfter(staticNodes);
-      },
+    inherits(babel) {
```

Comment:
can this be used in combination with enhanced classes? or is it using enhanced-classes or class-properties (where latter is ofc just a thin config wrapper over former)? could we validate misusage somehow?

Reply:
`class-properties` is meant to be a "switch" which turns on cass fields in the `enhanced-classes` plugin. If `enhanced-classes` hasn't already been added by the user, `class-properties` will do it.

Reply:
But can they be used side by side? Would `enhanced-classes` turned on by `class-properties` be "merged" with "standalone" `enhanced-classes`? In other words - is `enhanced-classes`' instance shareable among such invocations?